source/nogan/nogan.js

```diff
@@ -636,6 +652,18 @@ const getFlippedTiming = (timing) => {
 	}
 }
 
+/**
+ * Returns true if the peak is final.
+ * In other words, if it wouldn't be changed by any further pulses.
+ * This is specific to Arroost, not nogan.
+ * It's done so that we can stop iterating through wires that wouldn't change anything.
+ * @param {Peak} peak
+ * @returns
+ */
+const isPeakFinal = (peak) => {
```

Comment:
dodgy. maybe just remove this and revisit when behaviour is re-implemented

Reply:
It could be implemented by the behave!!!!! like as a secondary function. or like, could return false: true on the peak maybe